api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -495,17 +492,41 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
 
   @Override
   public ResponseEntity<WorkspaceResponseListResponse> getWorkspaces() {
-    // TODO: use FireCloud to determine what workspaces to return, instead of just returning
-    // workspaces from our database.
     User user = userProvider.get();
+    // This code is to migrate firecloud uuid to current workspaces
+    // once we are confident this has been done we can rework/remove
+    // much of this logic
     List<WorkspaceResponse> responseList = new ArrayList<WorkspaceResponse>();
     if (user != null) {
+      List<org.pmiops.workbench.firecloud.model.WorkspaceResponse> fcWorkspaces;
+      try {
+        fcWorkspaces = fireCloudService.getWorkspaces();
+      } catch (org.pmiops.workbench.firecloud.ApiException e) {
+        throw ExceptionUtils.convertFirecloudException(e);
+      }
       for (WorkspaceUserRole userRole : user.getWorkspaceUserRoles()) {
-        // TODO: Use FireCloud to determine access roles, not our DB
-        WorkspaceResponse currentWorkspace = new WorkspaceResponse();
-        currentWorkspace.setWorkspace(TO_CLIENT_WORKSPACE.apply(userRole.getWorkspace()));
-        currentWorkspace.setAccessLevel(userRole.getRole());
-        responseList.add(currentWorkspace);
+        org.pmiops.workbench.firecloud.model.WorkspaceResponse fcWorkspace;
+        org.pmiops.workbench.db.model.Workspace dbWorkspace;
+        dbWorkspace = workspaceService.getRequired(userRole.getWorkspace().getWorkspaceNamespace(), userRole.getWorkspace().getFirecloudName());
+        fcWorkspace = fcWorkspaces.stream().filter(w -> w.getWorkspace().getName().equals(TO_CLIENT_WORKSPACE.apply(userRole.getWorkspace()).getId())).collect(Collectors.toList()).get(0);
+        if (fcWorkspace == null) {
+          //  remove from our database
+          workspaceService.getDao().delete(dbWorkspace);
+        } else {
+          if (userRole.getWorkspace().getFirecloudUuid() == null) {
+            // populate UUID
+            dbWorkspace.setFirecloudUuid(fcWorkspace.getWorkspace().getWorkspaceId());
+            workspaceService.getDao().save(dbWorkspace);
```

Comment:
Did we want to save a tag / attribute in FC on these workspaces while we're at it? (Or are we going to do that later once there's something immutable we can set?)